compiler/src/yul/mappers/expressions.rs

```diff
@@ -239,7 +239,10 @@ pub fn expr_bin_operation(
             .typ;
 
         return match op.node {
-            fe::BinOperator::Add => Ok(expression! { add([yul_left], [yul_right]) }),
+            fe::BinOperator::Add => match typ.is_signed_integer() {
```

Comment:
i imagine at some point we'll add more checking functions for smaller integers?

Reply:
Ah, good point. I will expand this PR to include checks for all integer sizes for additions.

Reply:
Ok, done. Additions for all supported sizes should now be checked.